stablehlo/reference/Api.cpp

```diff
@@ -127,5 +132,26 @@ llvm::ErrorOr<SmallVector<InterpreterValue>> evalModule(
   return stablehlo::eval(mainFunc.getBody(), inputs, &fallback);
 }
 
+llvm::ErrorOr<OwningOpRef<ModuleOp>> parseModule(const std::string &mlir,
+                                                 MLIRContext &context) {
+  llvm::SourceMgr source_mgr;
+  source_mgr.AddNewSourceBuffer(llvm::MemoryBuffer::getMemBuffer(mlir),
+                                llvm::SMLoc());
+
+  mlir::DialectRegistry registry;
+  mlir::registerAllDialects(registry);
+  mlir::stablehlo::registerAllDialects(registry);
+  context.loadDialect<mlir::stablehlo::interpreter::InterpreterDialect>();
+  context.appendDialectRegistry(registry);
+
+  mlir::OwningOpRef<mlir::ModuleOp> module(
+      mlir::parseSourceFile<mlir::ModuleOp>(source_mgr, &context));
+
+  if (!module)
+    llvm::report_fatal_error("Failed to parse StableHLO module.");
```

Comment:
Return an llvm error instead? API already is `ErrorOr`, no need to fail fatally.